### DISCUSSION TITLE
Get the last visited url path

### DISCUSSION THREAD

**Discussion by ALEZ-DEV** [leptos-rs/leptos#2419] [Q&A]
I want to get the last url path the user has visited, like that if the user get the error page, I can add a button to go back to the previous page, is this possible without making a global variable ?

**Comment by gbj** [comment]
I can think of several good approaches to explore:
1. Create an event listener (using `window_event_listener` for example) that listens for the `popstate` event and keeps track of them.)
2. Use `use_location` to get reactive access to the current URL, and `create_effect` to store those values somewhere. 
3. If you literally just want to go back, you could use [`History::back`](https://docs.rs/web-sys/latest/web_sys/struct.History.html#method.back)

I haven't tried either of these three, although I know 2 will definitely work. Happy to help debug at some point.

**Reply by ALEZ-DEV** [reply]
thank I will try these solution
